feat: button 구현

LeeSuKyeong opened the pull request:
- 공통 버튼을 사용하도록 변경
- 버튼에 필요한 데이터 추가
- 수식 영역 svg 추가
- svg 파일 리팩토링
- 템플릿 영역 svg 추가

LeeSuKyeong committed:
modify: 버튼 컴포넌트 변경

- 수식 영역의 버튼을 common에 있는 버튼으로 변경

LeeSuKyeong committed:
style: oveflow시 scroll 생성

- overflow 시 scroll 생성하도록 변경

LeeSuKyeong committed:
feat: template 영역 버튼 구현

- template 영역에도 동일한 버튼 컴포넌트 추가
- svg 파일과 데이터  일부 구현

LeeSuKyeong committed:
feat: 수식 영역 svg 파일 추가

LeeSuKyeong committed:
modify: svg 파일 변경

LeeSuKyeong committed:
fix: 파일명 오타 수정

- 파일명 오타를 수정

LeeSuKyeong committed:
feat: latex data 추가

수식 영역에 필요한 latex data 추가

thdwlsgus0 commented on src/components/LeftSection/MathSection/style.tsx:
```diff
@@ -7,15 +7,11 @@ export const MathSectionContainer = styled.div`
+export const ButtonContainer = styled.div`
+  display: flex;
+  height: 80%;
+  flex-wrap: wrap;
```
flex-wrap이 설마 여러 행으로 만들어 주는 부분인가요..?

LeeSuKyeong replied:
지정된 width 초과되면 flex item이 다음줄로 보이게 되요

thdwlsgus0 replied:
앗.. 리팩토링할 때 참고하겠습니다!!

thdwlsgus0 commented on src/components/LeftSection/TemplateSection/index.tsx:
```diff
@@ -1,12 +1,40 @@
         <Title title="템플릿" />
+        <StyledComponent.ButtonContainer>
+          {math.map((data) => (
+            <LatexContent
```
LatexContent 재사용하신거 굿굿입니다.

thdwlsgus0 commented on src/utils/svg/leftsection/mathsection/large-operator/index.ts:
```diff
@@ -0,0 +1,28 @@
+import INTERSECTION from './intersection.svg';
```
이 방식 굿굿입니당

thdwlsgus0 reviewed (commented):
고생하셨습니다.. 한땀한땀.. 만들어 주는게 정말 힘드셨죠? ㅠㅠㅠ

thdwlsgus0 commented on src/components/LeftSection/TemplateSection/index.tsx:
```diff
@@ -1,12 +1,40 @@
 import * as StyledComponent from './style';
 
 const TemplateSectionContainer = () => {
+  // 임시로 변수 지정. 추후 props로 inputselected 받아와서 변경하도록 할예정
```
inputselected가 입력 수식 부분 말씀하시는거죠?

LeeSuKyeong replied:
네!

thdwlsgus0 replied:
제 부분 구현할 때 inputselected를 interface 타입으로 추가해야하나요?

LeeSuKyeong replied:
저도 이부분 고민인데 input 영역에서 click 한 버튼 값으로 한군데서 constants에 접근해서 매핑해서 리스트 가져와야해요
props로 리스트를 받는거랑 click action으로 상태변경해주면 상태값을 이 파일에서 useSelector로 가져와서 매핑한 리스트 가져오는 것 중에 뭐가 나을까요

thdwlsgus0 replied:
렌더링 측면에서 봤을 때 props로 하게 되면 부모 -> 자식 둘 다 렌더링이 일어나는 반면에 useSelector를 이용하게 되면 일부 상태면 변화했을 때만 컴포넌트가 리렌더링되니까 제 생각에는 useSelector가 더 나아보입니다. 

참고 링크 있습니다.
구현할 때 이 링크 참고하시면 될 것 같습니다.

https://react.vlpt.us/redux/08-optimize-useSelector.html
https://ooeunz.tistory.com/135

LeeSuKyeong committed:
feat: template 영역 svg 파일 추가

- tmplate 영역에 표시될 svg 파일을 추가
- tmplate 영역 dump data를 constant/templatesection로 변경
